services/QuillLMS/spec/services/analytics/segment_analytics_spec.rb

```diff
@@ -70,19 +70,28 @@
 
   context 'tracking activity completion' do
     let(:teacher) { create(:teacher) }
+    let(:topic) { create(:topic, level: 1) }
+    let(:second_topic) { create(:topic, level: 1) }
     let(:activity) { create(:diagnostic_activity) }
     let(:student) { create(:student) }
     let(:activity_session) { create(:activity_session) }
 
     it 'sends an event with information about the activity' do
+      create(:activity_topic, topic: topic, activity: activity)
```

Comment:
The conventional RSpec pattern is to create/initialize everything in `let` and `before` block _before_ the `it` block where the test is run.  Therefore, move 80-81 into a before block:

```
  before do
    create(:activity_topic, topic: topic1, activity: activity)
    create(:activity_topic, topic: topic2, activity: activity)
  end
```